pkg/controller/eventinstance/eventinstance.go

```diff
@@ -0,0 +1,105 @@
+package eventinstance
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+	"github.com/acorn-io/acorn/pkg/config"
+	"github.com/acorn-io/baaah/pkg/router"
+	"golang.org/x/sync/semaphore"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const defaultTTL = 168 * time.Hour // 7 days
+
+type Handler struct {
+	sem    *semaphore.Weighted
+	rawTTL string
+	ttl    time.Duration
+}
+
+func NewHandler() *Handler {
+	return &Handler{
+		sem: semaphore.NewWeighted(1),
+	}
+}
+
+func (h *Handler) UpdateTTL(req router.Request, _ router.Response) error {
+	if err := h.sem.Acquire(req.Ctx, 1); err != nil {
+		return fmt.Errorf("failed to acquire TTL semaphore for update: %w", err)
+	}
+	defer h.sem.Release(1)
+
+	cfg, err := config.Get(req.Ctx, req.Client)
```

Comment:
I think you can fetch the config from req.Object without having to do an extra api call

Reply:
pretty sure this is canonical way to get a config. we do it all over the place

Reply:
yeah but I guess then you will have to use another wrapper to convert con... probably not worth it

Reply:
So I switched this back. This call hits the cache most of the time and also does some required merging to get the correct config values.